lib/accession/accession/submission.rb

```diff
@@ -89,7 +89,11 @@ def close!
     private
 
     def check_sample
-      sample.errors.each { |key, value| errors.add key, value } unless sample.valid?
+      if sample.errors.is_a?(ActiveModel::Errors)
```

Comment:
Again. What is the reason for the switch? Before we were checking if the sample is valid which is the standard way to do it whereas here you are checking if there are errors. Does sample valid no longer work?

Reply:
We should remove these is_a? checks. Use the new way of adding errors.